### ISSUE TITLE
Support for custom `server` directive using `default_server` broken by 9b4bb07

### ISSUE THREAD

**Issue by Enchiridion** [nginx-proxy/nginx-proxy#2212]
I run several proxied sites simultaneously, to make it easier to see which sites are active I'm using docker-gen with a custom .tmpl file to auto-generate an html file, with links and tools for each site, whenever sites are started/stopped.  I'm displaying the page at http://localhost/, to do so my custom nginx config takes over being the default server.  I don't override `/etc/nginx/conf.d/default.conf`, I named mine so it loads first.  My custom config is really small:

```
/etc/nginx/conf.d/01-custom.conf

server {
  server_name __; # This is just an invalid value which will never trigger on a real hostname.
  listen 80 default_server;
  access_log /var/log/nginx/access.log;
  root /app/web/public;
  index index.htm;
}
```

An update in commit https://github.com/nginx-proxy/nginx-proxy/commit/9b4bb07b348dc5a428b94416517291adb30794c3 broke it because it added `default_server` on lines 406, 408, 412, and 414.  Nginx throws an error if there are multiple default servers for the same port.

From what I've read in the nginx docs, adding `default_server` is unnecessary in default.conf since the first `server` block becomes the default and `default_server` is only needed if you want to override that, which is what I need to do.  I don't know enough about the codebase of nginx-proxy to know if removing those now would be ok or not, else I'd make a PR to remove them.

In the meantime, I've made a copy of `nginx.tmpl`, removed default_server, and replaced it in container using my Dockerfile, but it's not a long-term solution.

**Comment by buchdag** [comment]
@rhansen could you take a look at this ?

**Comment by buchdag** [comment]
@Enchiridion I suspect you're using a method to get a default virtual host that previously only worked "by accident".

The correct, tested method is using [`DEFAULT_HOST`](https://github.com/nginx-proxy/nginx-proxy#default-host)

**Comment by rhansen** [comment]
While users could just run a new container with `DEFAULT_HOST` pointed at that, it seems reasonable to me to support users that want a custom `server` directive, not a container, as the fallback handler.  I'll open a PR with a fix and test coverage.

**Comment by Enchiridion** [comment]
Thanks for looking into and fixing it @buchdag and @rhansen !

**Comment by buchdag** [comment]
@Enchiridion could you try the fix proposed by @rhansen in #2214 and tell us if it works for you ?
